contracts/CourtRoom.sol

```diff
@@ -116,19 +150,20 @@ contract SwearGame is Owned {
 	function makeJudgement(bytes32 _claimId) private returns(bool) {
 
 	  bool claimantCompensated;
+    bool decision = false;
 
     require(caseContract.getStatus(_claimId) != 0);
 		// Somehow come to a resolution...
-		var(plaintiff,valid) = caseContract.getClaim(_claimId);
+		var(plaintiff,valid,beneficiary,blockNumber,sig_v,sig_r,sig_s) = caseContract.getClaim(_claimId);
 
 		// Case has already been compensated for
 		require(!valid);
 
-	  bool decision = takeDecision();
+    decision = (validatePromise(beneficiary,blockNumber,sig_v,sig_r,sig_s) && takeDecision());
```

Comment:
so here validatePromise should just be a generic call to a delegated contract which simply decides if player has a valid/active service agreement.